src/java.base/share/classes/javax/crypto/KDFSpi.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.crypto;
+
+import jdk.internal.javac.PreviewFeature;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KDFParameters;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+
+/**
+ * This class defines the <i>Service Provider Interface</i> (<b>SPI</b>) for the
+ * {@code KDF} class.
+ * <p>
+ * All the abstract methods in this class must be implemented by each
+ * cryptographic service provider who wishes to supply the implementation of a
+ * particular key derivation algorithm.
+ * <p>
+ * In addition, all implementations must provide a public constructor which
+ * accepts a {@code KDFParameters} object. The constructor must call {@code
+ * super(params)} passing the parameters supplied. The constructor must also
+ * throw an {@code InvalidAlgorithmParameterException} if the supplied
+ * parameters are inappropriate.
+ * <p>
+ * Implementations which do not support {@code KDFParameters} may require
+ * {@code null} to be passed, otherwise an {@code InvalidAlgorithmParameterException}
+ * may be thrown. On the other hand, implementations which require
+ * {@code KDFParameters} may throw an {@code InvalidAlgorithmParameterException}
+ * upon receiving a {@code null} value. Furthermore, implementations
+ * may supply default values for {@code KDFParameters}, mutating the
```

Comment:
The `mutating` word is suspicious. The object is very likely to be immutable. Just say a different object should be returned in the next sentence.

Reply:
OK, maybe you can just say the "actual" object. I still think `mutating` is strange.

Reply:
Personally I don't like the "mutate" word. But your sentence is quite complete now, so I don't have a strong opinion now.

Reply:
`KDFParameter` objects should ideally always be immutable (we should never provide mutable methods on a standard `KDFParameter` subclass), so I agree the word "mutating" is not right, what is being returned in this case is always going to be a new object.